contrib/cirrus/setup_environment.sh

```diff
@@ -19,71 +26,67 @@ do
     fi
 done
 
-# Setup env. vars common to all tasks/scripts/platforms and
-# ensure they return for every following script execution.
-MARK="# Added by $0, manual changes will be lost."
-touch "$HOME/$ENVLIB"
-if ! grep -q "$MARK" "$HOME/$ENVLIB"
-then
-    cp "$HOME/$ENVLIB" "$HOME/${ENVLIB}_original"
-    # N/B: Single-quote items evaluated every time, double-quotes only once (right now).
-    for envstr in \
-        "$MARK" \
-        "export EPOCH_TEST_COMMIT=\"$CIRRUS_BASE_SHA\"" \
-        "export HEAD=\"$CIRRUS_CHANGE_IN_REPO\"" \
-        "export TRAVIS=\"1\"" \
-        "export GOSRC=\"$CIRRUS_WORKING_DIR\"" \
-        "export OS_RELEASE_ID=\"$(os_release_id)\"" \
-        "export OS_RELEASE_VER=\"$(os_release_ver)\"" \
-        "export OS_REL_VER=\"$(os_release_id)-$(os_release_ver)\"" \
-        "export TEST_REMOTE_CLIENT=\"$TEST_REMOTE_CLIENT\"" \
-        "export BUILT_IMAGE_SUFFIX=\"-$CIRRUS_REPO_NAME-${CIRRUS_CHANGE_IN_REPO:0:8}\"" \
-        "export GOPATH=\"/var/tmp/go\"" \
-        'export PATH="$HOME/bin:$GOPATH/bin:/usr/local/bin:$PATH"' \
-        'export LD_LIBRARY_PATH="/usr/local/lib${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}"'
-    do
-        # Make permanent in later shells, and set in current shell
-        X=$(echo "$envstr" | tee -a "$HOME/$ENVLIB") && eval "$X" && echo "$X"
-    done
+# Sometimes environment setup needs to vary between distros
+# Note: This should only be used for environment variables, and temporary workarounds.
+#       Anything externally dependent, should be made fixed-in-time by adding to
+#       contrib/cirrus/packer/*_setup.sh to be incorporated into VM cache-images
+#       (see docs).
+case "${OS_REL_VER}" in
+    ubuntu-18) ;;
+    fedora-29)
+        # I have no idea why this sometimes happens
+        REMAINING=$(df /dev/sda1 | tail -1 | awk '{print $4}')
```

Comment:
There are a lot of assumptions here. I don't understand this CI environment enough to know if they're valid, and the next maintainer might not either, so could you add some comments describing:
* why `/dev/sda1` is guaranteed to be the root fs; and
* where the 100GB below comes from (preferably with pointers to other CI configs - I couldn't find where that's set); and
* where it's documented that the filesystem is e[234]fs; and
* how we know that `growpart` and `resize2fs` are guaranteed to be installed on the system?

Also, even though the commit comment says what "this sometimes happens" means, as a courtesy to future maintainers, could you describe it again? And perhaps as a bonus, in your error message, display $REMAINING?

Reply:
Yep, happy to.  For your own understanding:

Disk devices depends on the cloud architecture, we're not changing clouds anytime soon and Cirrus doesn't support adding additional disks/devices.

Disk-space is set at 200gig always (in `.cirrus.yml`), google's tooling complains if you try to use smaller, so 100gig is a safe bet.  If it's not a safe bet, I want jobs to fail and someone to tell me.

Growpart/resize are guaranteed since all base-images are the 'cloud' variants: They must contain `cloud-init` which depends on being able to handle this very resizing problem (naturally google doesn't use cloud-init).

The filesystem type will also depend on the base-image used, which is why this behavior is distro and version dependent.

I'm hoping to figure this problem out before anyone is able to read the comment :smile:

Reply:
Okay, I got this all commented.  I'll wait to push in case you find something else worth another test-run.